skimage/feature/blob.py

```diff
@@ -200,9 +200,11 @@ def blob_dog(image, min_sigma=1, max_sigma=50, sigma_ratio=1.6, threshold=2.0,
                                   footprint=np.ones((3, 3, 3)),
                                   threshold_rel=0.0,
                                   exclude_border=False)
-
+    # Convert local_maxima to float64
+    lm = local_maxima.astype(np.float64)
```

Comment:
Because then, `local_maxima[:, 2]` won't work as an indexing array in the next line,
```
local_maxima[:, 2] = sigma_list[local_maxima[:, 2]]
```
Another solution would of course be to extract the index first:
```
index = local_maxima[:, 2]
local_maxima = local_maxima.astype(np.float64)
local_maxima[:, 2] = sigma_list[index]
```
Another solution only takes two lines. However, this involves two copy operations instead of one, and is in my opinion much less readable:
```
local_maxima = local_maxima.astype(np.float64)
local_maxima[:, 2] = sigma_list[local_maxima[:, 2].astype(np.int_)]
```